llama_index/indices/vector_store/retrievers/retriever.py

```diff
@@ -81,7 +81,9 @@ def _retrieve(
             alpha=self._alpha,
             filters=self._filters,
         )
-        query_result = self._vector_store.query(query, **self._kwargs)
+
+        existing_doc_ids = set(self._index.index_struct.nodes_dict.keys())
+        query_result = self._vector_store.query(query, existing_doc_ids, **self._kwargs)
```

Comment:
I don't think this will work for external vector stores that stores text. We don't store node reference in the index struct for those. (Ya I know, quite suprising)